roachtest: add INSPECT performance benchmarks

spilchen opened the pull request:
Add two nightly roachtest benchmarks to measure INSPECT throughput across different test durations:
- Short run: 12 nodes × 8 CPUs, 400M rows, 1 index (~1 hour)
- Long run: 12 nodes × 8 CPUs, 600M rows, 2 indexes (~5 hours)

These configurations ensure that the number of spans exceeds the total CPU count (96), which is important because INSPECT parallelism is based on the number of spans.

Informs #154457

Epic: CRDB-30356

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/154956)
<!-- Reviewable:end -->

fqazi reviewed (approved):
@spilchen This looks great, only one minor comment.

> @fqazi reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154956#-:-Ob8VXRxDIZQkWUPrRaE:br6stws)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @spilchen)

___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 288 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/154956#-Ob8VIrB9bK4b88w0dp0:-Ob8VIrB9bK4b88w0dp1:b-plhgie) ([raw file](https://github.com/cockroachdb/cockroach/blob/8fc53e15aa6bf1b34d51af9cd12ea835d95912a0/pkg/cmd/roachtest/tests/inspect_throughput.go#L288)):*
> ```Go
> 		// The error contains "SET unsafe_setting_interlock_key = '<key>';"
> 		interlockKey := ""
> 		if strings.Contains(err.Error(), "SET unsafe_setting_interlock_key") {
> ```

Might be slightly safer to use logic similar to: https://github.com/cockroachdb/cockroach/blob/5f6ac15985b2fc404aeae7227b5f93fd540557d8/pkg/sql/tests/set_cluster_setting_interlock_test.go#L53


<!-- Sent from Reviewable.io -->

spilchen committed:
roachtest: add INSPECT performance benchmarks

Add two nightly roachtest benchmarks to measure INSPECT throughput
across different test durations:
- Short run: 12 nodes × 8 CPUs, 500M rows, 1 index (~1 hour)
- Long run: 12 nodes × 8 CPUs, 1B rows, 2 indexes (~5 hours)

These configurations ensure that the number of spans exceeds the total
CPU count (96), which is important because INSPECT parallelism is based
on the number of spans.

Informs #154457

Epic: CRDB-30356

Release note: none

spilchen reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/154956)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @fqazi)

___
*[`pkg/cmd/roachtest/tests/inspect_throughput.go` line 288 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/154956#-Ob8VIrB9bK4b88w0dp0:-Ob8ultsCRkytVsxe_fd:b-kakxm9) ([raw file](https://github.com/cockroachdb/cockroach/blob/8fc53e15aa6bf1b34d51af9cd12ea835d95912a0/pkg/cmd/roachtest/tests/inspect_throughput.go#L288)):*
<details><summary><i>Previously, fqazi (Faizan Qazi) wrote…</i></summary><blockquote>

Might be slightly safer to use logic similar to: https://github.com/cockroachdb/cockroach/blob/5f6ac15985b2fc404aeae7227b5f93fd540557d8/pkg/sql/tests/set_cluster_setting_interlock_test.go#L53
</blockquote></details>

Done. Thanks I was looking for a pattern on how to automate usage of these unsafe settings.


<!-- Sent from Reviewable.io -->
